Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Payment.hs

```diff
@@ -264,28 +264,39 @@ pdnNotificationStatus (_, merchantId, _) notificationId = do
   pdnNotification <- QNTF.findById notificationId >>= fromMaybeM (InternalError $ "No Notification Sent With Id" <> notificationId.getId)
   resp <- Payment.mandateNotificationStatus merchantId (mkNotificationRequest pdnNotification.shortId)
   let [responseCode, reponseMessage] = map (\func -> func =<< resp.providerResponse) [(.responseCode), (.responseMessage)]
-  processNotification pdnNotification.shortId resp.status responseCode reponseMessage
+  processNotification pdnNotification.shortId resp.status responseCode reponseMessage False
   return resp
   where
     mkNotificationRequest shortNotificationId =
       DPayments.NotificationStatusReq
         { notificationId = shortNotificationId
         }
 
-processNotification :: (CacheFlow m r, EsqDBFlow m r, EncFlow m r) => Text -> Payment.NotificationStatus -> Maybe Text -> Maybe Text -> m ()
-processNotification notificationId notificationStatus respCode respMessage = do
+processNotification :: (CacheFlow m r, EsqDBFlow m r, EncFlow m r) => Text -> Payment.NotificationStatus -> Maybe Text -> Maybe Text -> Bool -> m ()
+processNotification notificationId notificationStatus respCode respMessage fromWebhook = do
   notification <- QNTF.findByShortId notificationId >>= fromMaybeM (InternalError "notification not found")
   let driverFeeId = notification.driverFeeId
   unless (notification.status == Juspay.SUCCESS) $ do
-    driverFee <- QDF.findById driverFeeId
+    driverFee <- QDF.findById driverFeeId >>= fromMaybeM (InternalError $ "driver fee does not exist with id :" <> driverFeeId.getId)
```

Comment:
We already have error type defined for this, please use that. If not, define one.

Reply:
Sure